Document contributing (#2675)
agrc/gis.utah.gov: .github/contributing.md

for getting things done and landing your contribution.

1. Click the fork button in the top right to clone the [UGRC website Repository](https://github.com/agrc/gis.utah.gov/fork)

2. Clone your fork using SSH, GitHub CLI, or HTTPS.

   ```bash
   git clone git@github.com:<YOUR_GITHUB_USERNAME>/gis.utah.gov.git # SSH
   git clone https://github.com/<YOUR_GITHUB_USERNAME>/gis.utah.gov.git # HTTPS
   gh repo clone <YOUR_GITHUB_USERNAME>/gis.utah.gov # GitHub CLI
   ```

3. Change into the gis.utah.gov directory.

   ```bash
   cd gis.utah.gov
   ```

4. Create a remote to keep your fork and local clone up-to-date.

   ```bash
   git remote add upstream git@github.com:agrc/gis.utah.gov.git # SSH
   git remote add upstream https://github.com/agrc/gis.utah.gov.git # HTTPS
   gh repo sync agrc/gis.utah.gov # GitHub CLI
   ```

5. Create a new branch for your work.

   ```bash
   git checkout -b name-of-your-branch
   ```

6. Run the following to install the dependencies and start a local preview of your work.

   ```bash
   npm ci # installs this project's dependencies
   npm start # starts a preview of your local changes
   ```

> 7. Perform your changes. In case you're unfamiliar with the structure of this repository, we recommend a read on the [Collaborator Guide](./collaborating.md)

Reviewer: ```suggestion
7. Perform your changes. In case you're unfamiliar with the structure of this repository, we recommend that you review the [Collaborator Guide](.github/collaborating.md).
```

8. Perform a merge to sync your current branch with the upstream branch.

   ```bash
   git fetch upstream

>    git merge upstream/main

Reviewer: Why not rebase? Does a merge create less confusion?

Author: I wonder if merge queues will do the rebase and remove the merge commits. We should have been using the merge queues.

Reviewer: Yes, after reading that merge queue blog post for the third time, I'm finally getting a handle on them and they would have been very helpful over the past few weeks.

Reviewer: I think that maybe you could just get rid of this step and let merge queues handle it?

Author: Yeah possible. I just enabled them so let's see how it goes.

   ```

9. Run `npm run format` to confirm that linting and formatting are passing.

   ```bash
   npm run format
   ```

10. Once you're happy with your changes, add and commit them to your branch, then push the branch to your fork.

    ```bash
    cd ~/gis.utah.gov
    git add .
    git commit -m "some message"
    git push -u origin name-of-your-branch
    ```

    > [!IMPORTANT]\
    > Before committing and opening a Pull Request, please go first through our [Commit](#commit-guidelines) and [Pull Request](#pull-request-policy) guidelines outlined below.

11. Create a Pull Request.

> [!NOTE]\

> > We ask for PR authors to avoid to rebase/update their PRs with the base branch (`main`) unnecessarily.

Reviewer: ```suggestion
> We ask that PR authors avoid rebasing/updating their PRs with the base branch (`main`) unnecessarily.
```

> We use [GitHub Merge Queues](https://github.blog/2023-07-12-github-merge-queue-is-generally-available/)
> which means that before merge the PRs get automatically updated and checked against the latest changes on the base branch.
>

> > This also reduces the amount of times we need to run our CI checks, as every new push requires freshly new CI-checks.

Reviewer: ```suggestion
> This also reduces the amount of times we need to run our CI checks, as every new push requires fresh CI-checks.
```

    - This kind of PRs should only be done by existing collaborators with write-access and/or signed off by administrators/maintainers.
    - This rule cannot be used for updates on the `collaborating.md`, `CONTRIBUTING.md` guide, GitHub Actions, or any security-impacting file or document that changes the governing policies of this repository.
  - Pull requests might be "fast-tracked", meaning they can be merged before the usual 48 hours' notice if a "fast-track" label is added.
    - The person that is fast-tracking the PR (adding the label) must also comment on the PR that they're requesting the PR to be fast-tracked
    - Fast-tracking is only allowed for small bug fixes, small feature changes, localization changes, or other non-critical/highly-impacting changes not covered by the previous rule that allows PRs to be merged immediately.
    - Fast-tracking cannot be used for updates on the `collaborating.md`, `CONTRIBUTING.md` guide, GitHub Actions, or any security-impacting file or document that changes the governing policies of this repository.

> - There must be no objections after forty-eight (48) hours (Or seventy-two (72) hours if the PR was authored on the weekend).

Reviewer: This seems a little overkill to me. I'm curious where the idea came from. Just my two cents.

Author: It came from nodejs

Author: I agree it's a bit much. Happy to slash it down a bit
